src/HUD.cpp

```diff
@@ -0,0 +1,203 @@
+#include "HUD.h"
+#include "common.h"
+
+
+osg::Camera* createHUD(const std::string& logoFile, float scale, int winWidth,
+                       int winHeight)
+{
+    // --- HUD Camera ---
+    osg::Camera* hudCamera = new osg::Camera;
+
+    hudCamera->setReferenceFrame(osg::Transform::ABSOLUTE_RF);
+    hudCamera->setClearMask(GL_DEPTH_BUFFER_BIT);
+    hudCamera->setRenderOrder(osg::Camera::POST_RENDER);
+    hudCamera->setProjectionMatrix(
+        osg::Matrix::ortho2D(0, winWidth, 0, winHeight));
+    hudCamera->setViewMatrix(osg::Matrix::identity());
+    hudCamera->setAllowEventFocus(false);
+
+    // --- Load Texture ---
+    osg::ref_ptr<osg::Image> image = osgDB::readRefImageFile(logoFile);
+    if (!image)
+    {
+        osg::notify(osg::WARN)
+            << "Could not load HUD image: " << logoFile << std::endl;
+        return hudCamera;
+    }
+
+    osg::ref_ptr<osg::Texture2D> tex = new osg::Texture2D(image);
+
+    // --- Quad Geometry (top-right corner) ---
+    float w = image->s() * scale;
+    float h = image->t() * scale;
+    float x1 = winWidth - w - 20; // 20px margin from right
+    float y1 = winHeight - h - 20; // 20px margin from top
+    float x2 = x1 + w;
+    float y2 = y1 + h;
+
+    osg::ref_ptr<osg::Geometry> quad = new osg::Geometry;
+    osg::ref_ptr<osg::Vec3Array> verts = new osg::Vec3Array;
+
+    verts->push_back(osg::Vec3(x1, y1, 0));
+    verts->push_back(osg::Vec3(x2, y1, 0));
+    verts->push_back(osg::Vec3(x2, y2, 0));
+    verts->push_back(osg::Vec3(x1, y2, 0));
+
+    quad->setVertexArray(verts);
+
+    osg::ref_ptr<osg::Vec2Array> texcoords = new osg::Vec2Array;
+    texcoords->push_back(osg::Vec2(0, 0));
+    texcoords->push_back(osg::Vec2(1, 0));
+    texcoords->push_back(osg::Vec2(1, 1));
+    texcoords->push_back(osg::Vec2(0, 1));
+
+    quad->setTexCoordArray(0, texcoords);
+    quad->addPrimitiveSet(new osg::DrawArrays(GL_QUADS, 0, 4));
+
+    osg::ref_ptr<osg::Geode> geode = new osg::Geode();
+    geode->addDrawable(quad);
+
+    osg::ref_ptr<osg::StateSet> ss = geode->getOrCreateStateSet();
+    ss->setTextureAttributeAndModes(0, tex.get(), osg::StateAttribute::ON);
+    ss->setMode(GL_LIGHTING, osg::StateAttribute::OFF);
+    ss->setMode(GL_BLEND, osg::StateAttribute::ON);
+    ss->setRenderingHint(osg::StateSet::TRANSPARENT_BIN);
+    g_hudText = new osgText::Text;
+    g_hudText->setFont("font/OpenSans-VariableFont_wdth,wght.ttf");
+    g_hudText->setCharacterSize(28.0f);
+    g_hudText->setColor(osg::Vec4(1, 1, 1, 1));
+    g_hudText->setPosition(osg::Vec3(20, winHeight - 40, 0));

```

Comment:
Take a look at the picture: both the text and the logo have contrast issues. There are several ways to fix this. The easiest is to draw the elements multiple times with different colors and offsets, for example:

draw black text with offset vec2(1,1)
draw white text with offset vec2(0,0)

Alternatively, we could stick to single-pass rendering and use a shader to filter neighboring texels, which can also improve visibility without multiple draws.

![screen_shot_0_0](https://github.com/user-attachments/assets/00a30fb0-3d01-4628-a695-f637a95436d6)